### PR TITLE
chore: Update for 0.3.0 release, prepare for 0.4.0 development

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

Closes #.

### CODE REVIEW

**Review by parthchandra — APPROVED** [review]
lgtm
Do you have automation to change the version everywhere or is it mostly a manual search and replace?

**Comment by andygrove** [comment]
> lgtm Do you have automation to change the version everywhere or is it mostly a manual search and replace?

It is all manual at this point. It would be good to start automating this.

File: docs/source/user-guide/compatibility.md [hunk 1]
```diff
@@ -38,14 +38,6 @@ Comet uses the Rust regexp crate for evaluating regular expressions, and this ha
 regular expression engine. Comet will fall back to Spark for patterns that are known to produce different results, but
 this can be overridden by setting `spark.comet.regexp.allowIncompatible=true`.
 
-## Floating number comparison
```
Comment (kazuyukitanimura) [inline docs/source/user-guide/compatibility.md:41]: This change should be gone after rebasing and re-building?

**Review by kazuyukitanimura — APPROVED** [review]
Just one minor comment

File: docs/source/user-guide/compatibility.md (hunk 1 above)
Comment (andygrove) [reply docs/source/user-guide/compatibility.md]: Thanks. This is gone now.

**Pull request merged**